bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngineImpl.java

```diff
@@ -1123,9 +1123,27 @@ private void updateContext(String ruleUID, String moduleUID, Map<String, ?> outp
             for (Connection c : connections) {
                 String outputModuleId = c.getOuputModuleId();
                 if (outputModuleId != null) {
-                    sb.append(outputModuleId).append(OUTPUT_SEPARATOR).append(c.getOutputName());
-                    context.put(c.getInputName(), context.get(sb.toString()));
-                    sb.setLength(0);
+                    String outputName = c.getOutputName();
+                    sb.append(outputModuleId).append(OUTPUT_SEPARATOR);
+                    // if the outputName contains a dot we assume its a hashmap and retrieve the value after the dot
+                    if (outputName != null && outputName.contains(".")) {
+                        String[] parts = outputName.split("\\.");
+                        sb.append(parts[0]);
+                        Object dataStructure = context.get(sb.toString());
+                        if (dataStructure instanceof HashMap) {
```

Comment:
If the data structure is not an instance of a HashMap, shouldn't we keep the old behavior?

Reply:
The old behaviour is in the `else` branch in line 1142. The test for a `HashMap` is inside the `if (outputName != null && outputName.contains(".")) {` which means that the outputName has a "." in it which is not allowed in variable names anyway, so it either is a data structure (and I only want to allow a `HashMap`) or it is an invalid name.